### ISSUE TITLE
the RDA in this code does not use Distribution-balanced focal loss

### ISSUE THREAD

**Issue by zengtao568** [ailearn-ml/RDA#1]
I have some questions. The default value of weight set in the forward function in the resample_loss.py file is None.  ResampleLoss is initialized without setting weight. 
In the RDA.py file, self.loss_func(pred, y) is called like this, so the parameter weight is still not set.
In other words, weight will remain None. Even if reweight_func='rebalance' is set when initializing ResampleLoss, since weight is None, no reweighting operation will be performed.
After my testing, reweight_function() is indeed not used in the code. So the RDA in this code does not use Distribution-balanced focal loss.
I'm not sure my opinion is correct, please pay attention to this issue

**Comment by ailearn-ml** [comment]
> I have some questions. The default value of weight set in the forward function in the resample_loss.py file is None. ResampleLoss is initialized without setting weight. In the RDA.py file, self.loss_func(pred, y) is called like this, so the parameter weight is still not set. In other words, weight will remain None. Even if reweight_func='rebalance' is set when initializing ResampleLoss, since weight is None, no reweighting operation will be performed. After my testing, reweight_function() is indeed not used in the code. So the RDA in this code does not use Distribution-balanced focal loss. I'm not sure my opinion is correct, please pay attention to this issue

Dear zengtao568,

Thank you for your attention to our work. We have updated the code to the latest version where RDA correctly uses Distribution-balanced focal loss. Thank you very much!

**Comment by zengtao568** [comment]
Thank you very much for your attention to my question.
I have an additional suggestion that I didn't mention before. On line 62 of your run_RDA.py file,
`best_state_dict = model.state_dict()`
this assignment will cause your best_state_dict to change as model.state_dict() changes. So the best.tar is the model after running 300 epochs
I suggest you change it to this form: 
`best_state_dict = copy.deepcopy(model.state_dict())`
This may be the way you really think.
Through the experiment, the effect has been improved.

**Comment by ailearn-ml** [comment]
Many thanks for your valuable suggestion! ☺☺☺
